src/main/java/org/breedinginsight/brapps/importer/services/processors/ProcessorManager.java

```diff
@@ -114,6 +114,9 @@ private void postBrapiData(Program program, ImportUpload upload) throws Validato
             processor.postBrapiData(mappedBrAPIImport, program, upload);
         }
 
+        // Update progress to reflect final finished and inProgress counts.
+        upload.updateProgress(Math.toIntExact(totalObjects), 0);
```

Comment:
Could this be moved into the ImportStatusService::finishUpload method? Mainly just an organizational thing to keep all modifications to the ImportUpload object being done in there.

Reply:
Done and tested, b09ddeaf.